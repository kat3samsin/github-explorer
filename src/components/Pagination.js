import React, { Component } from 'react';
import { turnPage } from "../actions/getProjects";
import { connect } from 'react-redux';

export class Pagination extends Component {
  turnPage(page) {
    this.props.dispatch(turnPage(page));
  }

  render() {
    return this.props.totalPage === 1 ? (
      <div className='pagination'>
          Page {this.props.page} of {this.props.totalPage}&nbsp;&nbsp;
      </div>
    ) : (
        <div className='pagination'>
            Page {this.props.page} of {this.props.totalPage}&nbsp;&nbsp;
            <span className='link' onClick={this.turnPage.bind(this, 'first')}>&#60;&#60;</span>&nbsp;
            <span className='link' onClick={this.turnPage.bind(this, 'prev')}>&#60;</span>&nbsp;
            <span className='link' onClick={this.turnPage.bind(this, 'next')}>&#62;</span>&nbsp;
            <span className='link' onClick={this.turnPage.bind(this, 'last')}>&#62;&#62;</span>
        </div>
      );
   }
}

const mapStateToProps = (state) => {
  return state;
}

export default connect(mapStateToProps)(Pagination);