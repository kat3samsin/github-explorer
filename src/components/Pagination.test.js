import React from 'react';
import { shallow } from 'enzyme';
import { Pagination } from './Pagination';

describe('Pagination', () => {
  const turnPageFn = jest.fn();
  let wrapper;
  let page = 1;
  let totalPage = 5;

  beforeEach(() => {
    wrapper = shallow(<Pagination turnPage={turnPageFn} page={page} totalPage={totalPage} />)
  });
  afterEach(() => {
    turnPageFn.mockRestore();
  })

  it('renders without crashing', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('if result is only 1 page', () => {
    it('should not show navigation links', () => {
      let component = shallow(<Pagination turnPage={turnPageFn} page={1} totalPage={1} />)
      expect(component.html()).toBe('<div class=\"pagination\">Page 1 of 1</div>');
    });
  });

  describe('if result is 5 pages', () => {
    it('should show navigation links', () => {
      expect(wrapper.find('div').text()).toMatch('Page 1 of 5');
      expect(wrapper.find('span')).toHaveLength(4);
    });
  });

  describe('click first page', () => {
    it('should pass first', () => {
      wrapper.find('span').at(0).simulate('click');
      expect(turnPageFn).toHaveBeenCalledWith('first');
    });
  });

  describe('click next page', () => {
    it('should pass next', () => {
      wrapper.find('span').at(1).simulate('click');
      expect(turnPageFn).toHaveBeenCalledWith('prev');
    });
  });

  describe('click prev page', () => {
    it('should pass prev', () => {
      wrapper.find('span').at(2).simulate('click');
      expect(turnPageFn).toHaveBeenCalledWith('next');
    });
  });

  describe('click last page', () => {
    it('should last prev', () => {
      wrapper.find('span').at(3).simulate('click');
      expect(turnPageFn).toHaveBeenCalledWith('last');
    });
  });
});