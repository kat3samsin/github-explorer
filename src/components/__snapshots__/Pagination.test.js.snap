// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Pagination
    page={1}
    totalPage={5}
    turnPage={
      [MockFunction] {
        "calls": Array [
          Array [
            "first",
          ],
          Array [
            "prev",
          ],
          Array [
            "next",
          ],
          Array [
            "last",
          ],
        ],
        "results": Array [
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
          Object {
            "isThrow": false,
            "value": undefined,
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Page ",
        1,
        " of ",
        5,
        "  ",
        <span
          className="link"
          id="first"
        >
          &lt;&lt;
        </span>,
        " ",
        <span
          className="link"
          id="prev"
        >
          &lt;
        </span>,
        " ",
        <span
          className="link"
          id="next"
        >
          &gt;
        </span>,
        " ",
        <span
          className="link"
          id="last"
        >
          &gt;&gt;
        </span>,
      ],
      "className": "pagination",
    },
    "ref": null,
    "rendered": Array [
      "Page ",
      1,
      " of ",
      5,
      "  ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "<<",
          "className": "link",
          "id": "first",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "<<",
        "type": "span",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "<",
          "className": "link",
          "id": "prev",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "<",
        "type": "span",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": ">",
          "className": "link",
          "id": "next",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": ">",
        "type": "span",
      },
      " ",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": ">>",
          "className": "link",
          "id": "last",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": ">>",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Page ",
          1,
          " of ",
          5,
          "  ",
          <span
            className="link"
            id="first"
          >
            &lt;&lt;
          </span>,
          " ",
          <span
            className="link"
            id="prev"
          >
            &lt;
          </span>,
          " ",
          <span
            className="link"
            id="next"
          >
            &gt;
          </span>,
          " ",
          <span
            className="link"
            id="last"
          >
            &gt;&gt;
          </span>,
        ],
        "className": "pagination",
      },
      "ref": null,
      "rendered": Array [
        "Page ",
        1,
        " of ",
        5,
        "  ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "<<",
            "className": "link",
            "id": "first",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "<<",
          "type": "span",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "<",
            "className": "link",
            "id": "prev",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "<",
          "type": "span",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": ">",
            "className": "link",
            "id": "next",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": ">",
          "type": "span",
        },
        " ",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": ">>",
            "className": "link",
            "id": "last",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": ">>",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
